autogpt_platform/backend/backend/data/queue.py

```diff
@@ -16,35 +18,74 @@ def default(self, o):
         return super().default(o)
 
 
-class AbstractEventQueue(ABC):
-    @abstractmethod
-    def put(self, execution_result: ExecutionResult):
-        pass
+class BaseRedisEventBus:
```

Comment:
could you make this `Generic[T]` or rename it to reflect what kind of events it handles?

Reply:
I did the same here for the `RedisEventQueue`: https://github.com/Significant-Gravitas/AutoGPT/pull/8358/commits/a82acc77880962a0842a774707622d80cbe87e0f

Reply:
Updated. Are you using this event queue using different type by any chance on your PR?